nexus/src/app/background/tasks/inventory_load.rs

```diff
@@ -0,0 +1,264 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Background task for loading the latest inventory collection from the DB
+
+use crate::app::background::BackgroundTask;
+use chrono::Utc;
+use futures::future::BoxFuture;
+use nexus_auth::context::OpContext;
+use nexus_db_queries::db::DataStore;
+use nexus_types::internal_api::background::InventoryLoadStatus;
+use nexus_types::inventory::Collection;
+use serde_json::json;
+use slog_error_chain::InlineErrorChain;
+use std::sync::Arc;
+use tokio::sync::watch;
+
+pub struct InventoryLoader {
+    datastore: Arc<DataStore>,
+    tx: watch::Sender<Option<Arc<Collection>>>,
```

Comment:
note that a `watch::Receiver` is basically just "an `Arc` plus some other stuff", so depending on usage patterns, it _may_ not be necessary to have two layers of reference counting. however, this does let us hold onto prior inventories if they're still being used when a new one is loaded, so this may be desirable...